### PR TITLE
Adding e2e test for random beacon

### PR DESCRIPTION
**Pull request by dimpar**
Testing happy path with genesis and new relay requests. It should trigger dkg 3 times and create 3 groups in total. 

A group was selected by calculation `groupId = prevEntry % numberOfActiveGroups`

All group signatures were generated separately (in GO lang) with the appropriate group private key over the prev signature.

### CODE REVIEW

File: solidity/random-beacon/test/system/e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  const randomBeacon = contracts.randomBeacon as RandomBeaconStub & RandomBeacon
+  const testToken = contracts.testToken as TestToken
+
+  return {
+    randomBeacon,
+    testToken,
+  }
+}
+
+describe.only("System -- e2e", () => {
```
Comment (tomaszslabon) [inline solidity/random-beacon/test/system/e2e.test.ts:50]: `only` should probably be removed.

File: solidity/random-beacon/test/data/bls.ts [hunk 2]
```diff
@@ -2,10 +2,22 @@ const blsData = {
   // data generated using master secret key 123
   secretKey: 123,
 
-  // altbn128 public key for secret key 123
+  // altbn128 public key for secret key 123. For implemenation details refer to bls_test.go
```
Comment (tomaszslabon) [inline solidity/random-beacon/test/data/bls.ts:5]: implemenation => implementation

File: solidity/random-beacon/test/system/e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+  const randomBeacon = contracts.randomBeacon as RandomBeaconStub & RandomBeacon
+  const testToken = contracts.testToken as TestToken
+
+  return {
+    randomBeacon,
+    testToken,
+  }
+}
+
```
Comment (tomaszslabon) [inline solidity/random-beacon/test/system/e2e.test.ts:49]: Maybe some brief description of the test scenario?

File: solidity/random-beacon/test/data/bls.ts [hunk 4]
```diff
@@ -14,13 +26,43 @@ const blsData = {
...
+    // signed by group with 'groupPubKey2'
+    "0x08875f9e43e3498c4a139cc76e5311c530bbb589236213b7b668ce216b7a07902d280aaad20d2b091daa8dcc729d383e9da57482f07c32aee27acd8d71f45ba2",
+    // signed by group with 'groupPubKey2'
+    "0x15bbb764a647b078a37e96adc9f5eda5863d3b1fd0d027eabc15010da4f35afb07cb7e1660d5776f1ca91bbd255f79dc669a8197ce8589230f94f7a31f213029",
+    // signed by group with 'groupPubKey'
+    "0x0c62b62095142271bb6ed0474e2ce6299395a83d233c1b12da41d49e8516cfa80c27ff0ce575d5c686b0be9e68f8328a92056b8093c16741b8756df7a066c196",
+    // signed by group with 'groupPubKey3'
+    "0x25b12af46496104317227c716c85a971c2a74908ad43c2ca6d9ac830a6a3650f0ce79fb5dd2c18eeecbcbf5f586dea29aa09e69e97cc14e7f0e670fb8e7bc189",
+    // signed by group with 'groupPubKey1'
+    "0x245638c2a457cad293e140749d2ac45e851d73cc259f4202ac497238b68655c82ad7835b7431bf90d97890199d5354939494b8570b771013671bc9161e0c2406",
```
Comment (tomaszslabon) [inline solidity/random-beacon/test/data/bls.ts:56]: Are those `// signed by group with ...` correct?

File: solidity/random-beacon/test/system/e2e.test.ts (hunk 1 above)
Comment (dimpar) [reply solidity/random-beacon/test/system/e2e.test.ts]: removed

File: solidity/random-beacon/test/data/bls.ts (hunk 2 above)
Comment (dimpar) [reply solidity/random-beacon/test/data/bls.ts]: done d3e3969

File: solidity/random-beacon/test/system/e2e.test.ts (hunk 3 above)
Comment (dimpar) [reply solidity/random-beacon/test/system/e2e.test.ts:49]: added d3e3969

File: solidity/random-beacon/test/data/bls.ts (hunk 4 above)
Comment (dimpar) [reply solidity/random-beacon/test/data/bls.ts:56]: yes, I reworded a bit, but the general idea is that every 5th relay request a new group is added. The following equation `groupId = prevEntry % numberOfActiveGroups` determines which active group signs the prev entry. Group pub key is sort of a unique id and that's why I added comments to denote which group signed a prev relay.

**Comment by dimpar** [comment]
@tomaszslabon ready for another look

**Comment by tomaszslabon** [comment]
> @tomaszslabon ready for another look

@dimpar I have no further comments. As it's quite an important test, I would give a chance to other solidity developers to look at it (CC @pdyraga @nkuba @lukasz-zimnoch) before merging.

**Review by tomaszslabon — APPROVED** [review]

**Review by lukasz-zimnoch — APPROVED** [review]
Looks good!

**Pull request merged**